### PR TITLE
fix(stake): exclude realized_junior_loss from ReturnInsurance returnable cap

### LINKED ISSUE #269: ReturnInsurance can push total_returned above total_flushed and windfall senior with forfeited junior capital
## Summary

`process_return_insurance` can drive `total_returned` above `total_flushed` and windfall the senior tranche with capital that was already forfeited by an exited junior tranche. This re-opens the last-junior-exit windfall that the junior-loss settlement was written to prevent.

## Background: the last-junior-exit settlement

When the last junior LP exits while an insurance loss is outstanding, `process_withdraw` "realizes" the loss the junior absorbed (`L = junior_balance − effective_junior_balance`) by booking, in the same instruction:

```rust
pool.total_returned         += L;   // phantom settlement — NO tokens move
pool.set_realized_junior_loss(pool.realized_junior_loss() + L);
```

The phantom `+L` on `total_returned` and the `−realized_junior_loss` term in `total_pool_value()` cancel, so pool value is unchanged at exit and the protected senior tranche is not windfalled. The `realized_junior_loss` doc-comment states this portion must "sit as DEAD (unclaimable) value rather than windfalling senior," and the existing unit test `test_161_last_junior_exit_does_not_windfall_recovery_to_senior` asserts senior stays at its principal.

## The defect

`process_return_insurance` computes its returnable cap as:

```rust
let outstanding = pool.total_flushed
    .saturating_sub(pool.total_returned)
    .saturating_add(pool.realized_junior_loss());   // <-- re-exposes the dead portion
```

and then, after transferring `amount` tokens into the vault, does:

```rust
pool.total_returned = pool.total_returned.checked_add(amount)?;   // no realized_junior_loss decrement
```

Because the last-junior exit already incremented `total_returned` by the forfeited `L` (phantom), returning the realized portion here double-counts `total_returned` for that amount. Consequences:

1. `total_returned` exceeds `total_flushed`, breaking the documented conservation invariant.
2. The `−realized_junior_loss` term in `total_pool_value()` is canceled, so the dead forfeited-junior capital becomes claimable and flows entirely to the protected senior tranche.

The unit test does not catch this because it stops immediately after the exit and never exercises `ReturnInsurance`. The permissionless `RecoverFlushedInsurance` path is unaffected — its cap correctly excludes `realized_junior_loss` — so only the admin `ReturnInsurance` path (used in the normal recovery flow) triggers the windfall.

## Impact

A junior-absorbed loss followed by the last junior exiting and the admin performing the ordinary recovery flow (WithdrawInsurance on the wrapper, then `ReturnInsurance`) transfers the exited junior's forfeited first-loss capital to the senior tranche. This violates the tranche first-loss guarantee and the conservation invariant `total_returned <= total_flushed`. No external attacker is required — it occurs under normal admin operation. It is a value-misallocation between tranches (the tokens are physically present, so no protocol insolvency results).

## Proof of concept

The test below drives the same scenario as `test_161_last_junior_exit_does_not_windfall_recovery_to_senior` one step further into the real `ReturnInsurance` cap + booking. It uses the program's real accounting methods (`total_pool_value`, `senior_balance`, `effective_junior_balance`) and mirrors the two inline processor bookings verbatim (each cited). It asserts the invariants that must hold; on current `main` they fail.

`tests/poc_return_insurance_realized_junior_loss_windfall.rs`:

```rust
use bytemuck::Zeroable;
use percolator_stake::state::StakePool;

fn tranche_pool() -> StakePool {
    let mut p = StakePool::zeroed();
    p.is_initialized = 1;
    p.set_discriminator();
    p.set_tranche_enabled(true);
    p.set_junior_fee_mult_bps(20_000);
    p // pool_mode 0 (insurance LP)
}

#[test]
fn return_insurance_must_not_windfall_senior_with_realized_junior_loss() {
    let mut pool = tranche_pool();

    // Junior 100k + senior 100k (each mints 1:1 into its sub-pool).
    pool.total_deposited = 200_000;
    pool.total_lp_supply = 200_000;
    pool.set_junior_balance(100_000);
    pool.set_junior_total_lp(100_000);
    assert_eq!(pool.senior_balance().unwrap(), 100_000, "senior 100k pre-loss");
    assert_eq!(pool.effective_junior_balance(), 100_000, "junior 100k pre-loss");

    // Admin flushes 50k — a junior-absorbed loss (net_loss 50k <= junior 100k).
    pool.total_flushed = 50_000;
    assert_eq!(pool.effective_junior_balance(), 50_000, "junior marked down to 50k");
    assert_eq!(
        pool.senior_balance().unwrap(),
        100_000,
        "senior protected by junior first-loss"
    );

    // ── Last junior exits. Mirror process_withdraw's full-exit branch verbatim ──
    let junior_payout = pool.effective_junior_balance(); // 50k
    pool.total_withdrawn += junior_payout;
    pool.total_lp_supply -= 100_000;
    let net_loss = pool.total_flushed.saturating_sub(pool.total_returned);
    let forfeited = pool
        .junior_balance()
        .saturating_sub(pool.effective_junior_balance())
        .min(net_loss);
    assert_eq!(forfeited, 50_000, "junior forfeits the 50k loss it absorbed");
    pool.total_returned += forfeited; // phantom settlement — NO tokens moved
    pool.set_realized_junior_loss(pool.realized_junior_loss() + forfeited);
    pool.set_junior_total_lp(0);
    pool.set_junior_balance(0);

    // Invariant holds at exit: senior stays at principal; forfeited 50k is dead value.
    assert_eq!(pool.total_pool_value().unwrap(), 100_000, "tpv excludes dead forfeited loss");
    assert_eq!(pool.senior_balance().unwrap(), 100_000, "senior not windfalled at exit");
    assert_eq!(pool.total_returned, pool.total_flushed, "ledger loss settled at exit");
    assert_eq!(pool.realized_junior_loss(), 50_000);

    // ── Admin performs the NORMAL recovery flow. Mirror the ReturnInsurance cap verbatim ──
    let outstanding = pool
        .total_flushed
        .saturating_sub(pool.total_returned)
        .saturating_add(pool.realized_junior_loss());
    assert_eq!(outstanding, 50_000, "cap re-exposes the realized/dead 50k");

    // Admin returns the full outstanding amount (real tokens; total_returned += amount,
    // with NO decrement of realized_junior_loss — mirrors process_return_insurance).
    let amount = outstanding;
    pool.total_returned += amount;

    // ── Invariants that MUST hold (they FAIL on current code) ──
    assert!(
        pool.total_returned <= pool.total_flushed,
        "conservation broken: total_returned ({}) exceeds total_flushed ({})",
        pool.total_returned,
        pool.total_flushed
    );
    assert_eq!(
        pool.senior_balance().unwrap(),
        100_000,
        "senior must NOT be windfalled by the forfeited-junior recovery (got {})",
        pool.senior_balance().unwrap()
    );
}
```

### Observed failure on current `main`

```
running 1 test
test return_insurance_must_not_windfall_senior_with_realized_junior_loss ... FAILED

---- return_insurance_must_not_windfall_senior_with_realized_junior_loss stdout ----
thread 'return_insurance_must_not_windfall_senior_with_realized_junior_loss' panicked at
tests/poc_return_insurance_realized_junior_loss_windfall.rs:117:5:
conservation broken: total_returned (100000) exceeds total_flushed (50000)

test result: FAILED. 0 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out
```

After the conservation assertion, the senior-balance assertion would likewise fail: `senior_balance()` reports 150_000 instead of the protected 100_000 — the exited junior's forfeited 50k has flowed to senior.

## Suggested remediation

Make the `ReturnInsurance` returnable cap exclude `realized_junior_loss`, matching `RecoverFlushedInsurance`:

```rust
let outstanding = pool.total_flushed.saturating_sub(pool.total_returned);
```

This preserves `total_returned <= total_flushed`, keeps the forfeited junior portion permanently unrecoverable (as intended by the settlement design), and eliminates the senior windfall. The legitimate recoverable loss (`total_flushed − total_returned`) remains fully returnable, so no genuine recovery capacity is lost.

### PR DESCRIPTION
**Pull request by 0x-SquidSol**
## What

Fixes the `ReturnInsurance` returnable cap so it can no longer push `total_returned` above `total_flushed` or windfall the senior tranche with capital forfeited by an exited junior tranche.

Closes #269.

## Why

When the last junior LP exits during an outstanding insurance loss, `process_withdraw` settles the forfeited junior loss `L` with a phantom booking (`total_returned += L`) and records it in `realized_junior_loss` so `total_pool_value()` subtracts it — leaving that portion dead/unclaimable, which is what keeps the protected senior tranche from being windfalled (#161).

`process_return_insurance` computed its cap as `total_flushed − total_returned + realized_junior_loss`, adding the dead portion back. Since the handler then does `total_returned += amount` with no matching `realized_junior_loss` decrement, returning that portion double-counts `total_returned`:

- `total_returned` rises above `total_flushed` (conservation invariant broken), and
- the `−realized_junior_loss` term in `total_pool_value()` is canceled, so the forfeited junior capital flows to senior.

This occurs under the normal admin recovery flow (WithdrawInsurance on the wrapper, then `ReturnInsurance`) — no external attacker required. The tokens are physically present, so it is a value-misallocation between tranches rather than a protocol insolvency.

## Change

- Add `StakePool::insurance_recoverable()` = `total_flushed.saturating_sub(total_returned)` as the single source of truth for both returnable caps, so `ReturnInsurance` and `RecoverFlushedInsurance` can never diverge. It excludes `realized_junior_loss` (already settled at exit).
- Route `process_return_insurance` (previously the buggy `+ realized_junior_loss` cap) and `process_recover_flushed_insurance` through the helper.
- Update the `realized_junior_loss` doc-comment to accurately state the forfeited portion stays in the wrapper insurance fund and is not returnable to the vault.

The genuinely-recoverable loss (`total_flushed − total_returned`) remains fully returnable; only the buggy dead-portion capacity is removed.

## Test

Adds `tests/poc_return_insurance_realized_junior_loss_windfall.rs`, which drives the `#161` scenario one step further into the real returnable cap (`insurance_recoverable()`) and the return booking, then asserts `total_returned <= total_flushed` and that senior stays at its principal. It fails against the previous `+ realized_junior_loss` cap and passes with this change.

## Verification

- New regression test passes.
- `cargo test --lib` (incl. `test_161_last_junior_exit_does_not_windfall_recovery_to_senior`) passes.
- Existing tranche / insurance / accounting suites pass.

## Summary by CodeRabbit

* **Bug Fixes**
  * Corrected insurance recovery calculations to exclude losses already settled through junior-loss realization.
  * Prevented settled losses from being returned or counted again through either insurance recovery path.
  * Preserved senior tranche balances during realized-loss scenarios.

* **Tests**
  * Added regression coverage confirming recoverable insurance amounts, balance conservation, and protection against unexpected windfalls.

### CODE REVIEW

**Comment by dcccrypto** [comment]
Superseded by #279.

Your windfall analysis is right and #279 preserves that outcome: forfeited junior capital stays in the wrapper and senior ends at principal. Your doc/conservation rewrite was the better-written of the two competing PRs and I carried its reasoning across.

The blocker is that this cap can never satisfy the H-1 resolve gate. With `flushed=1000 / returned=300 (phantom) / realized_junior_loss=300`, `flushed − returned` tops out at 700, so `total_recovered_from_wrapper` never reaches `total_flushed` and the gate at `processor.rs:3553` / `:3640` rejects **permanently** — the market becomes unresolvable. Neither this PR nor #262 touches that gate. (The deployed code has the same defect, since it uses this same formula for Recover.)

#279 fixes both halves: cap and gate both go through `total_flushed − realized_junior_loss − total_recovered_from_wrapper`, which converges to 0 *and* is reachable, without windfalling senior.
